src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -0,0 +1,318 @@
+               (dilation_w != 1) || (pad_h != 0) || (pad_w != 0))
+            {
+                continue;
+            }
+        }
+        if((gemm_n % pConfig->gemm_n_per_block != 0) || (gemm_m % pConfig->gemm_m_per_block != 0))
+        {
+            continue;
+        }
+        if(pConfig->gemm_n_per_block % pConfig->nxb != 0)
+        {
+            continue;
+        }
+        //# ho * wo is 4x, gemm_n is 256, hence need batch size 256/4=64x
+        if(n % (pConfig->gemm_n_per_block / pConfig->nxb) != 0)
+        {
+            continue;
+        }
+        if((h_tilda_slice * w_tilda_slice) % pConfig->nxb != 0)
+        {
+            continue;
+        }
+        bool gemm_k_valid = true;
+        for(int gemm_id = 0; gemm_id < num_of_gemm; gemm_id++)
+        {
+            int i_y_tilda          = gemm_id / x_tilda;
+            int i_x_tilda          = gemm_id % x_tilda;
+            int y_dot_slice        = (i_y_tilda + 1) * y_dot <= y ? y_dot : y % y_dot;
+            int x_dot_slice        = (i_x_tilda + 1) * x_dot <= x ? x_dot : x % x_dot;
+            int gemm_k             = k * y_dot_slice * x_dot_slice;
+            bool is_gemm_not_empty = gemm_k > 0;
+            if(is_gemm_not_empty)
+            {
+                if(gemm_k % pConfig->gemm_k_per_block != 0)
+                    gemm_k_valid = false;
+            }
+        }
+        if(!gemm_k_valid)
+            continue;
+        // if all valid, break out;
+        break;
+    }
+    if(pConfig != tunables.end())
+    {
+        kernel_name = pConfig->GetKernelName();
+
+        block_size = pConfig->GetBlockSize();
+
+        grid_size = integer_divide_ceil(gemm_m, pConfig->gemm_m_per_block) *
+                    integer_divide_ceil(gemm_n, pConfig->gemm_n_per_block);
+        return true;
+    }
+    return false;
+}
+
+bool ConvAsmImplicitGemmGTCDynamicBwdXdlops::IsApplicable(const ConvolutionContext& ctx) const
+{
+    const auto device_name = ctx.GetStream().GetDeviceName();
+    if(!(StartsWith(device_name, "gfx908")))
+        return false;
+
+    if(!ctx.direction.IsBackwardData())
+        return false;
+
+    if(!ctx.Is2d())
+        return false;
+
+    if(!ctx.IsFp32())
+        return false;
+
+    if(!ctx.rmv.IsV3())
+        return false;
+
+    if(ctx.group_counts != 1)
+        return false;
+    std::string kernel_name;
+    int block_size;
+    int grid_size;
+    return FindImplicitGemmGtcDynamicBwdKernel(ctx, kernel_name, block_size, grid_size);
+}
+
+ConvSolution
+ConvAsmImplicitGemmGTCDynamicBwdXdlops::GetSolution(const ConvolutionContext& ctx) const
+{
+    ConvSolution result;
+    KernelInfo kernel;
+    std::ostringstream options;
+
+    std::string kernel_name;
+    int block_size;
+    int grid_size;
+    bool ret = FindImplicitGemmGtcDynamicBwdKernel(ctx, kernel_name, block_size, grid_size);
+    if(!ret)
+        MIOPEN_THROW("this kernel should not run with igemm dynamic!");
```

Comment:
This check is not necessary, please remove.

Reply:
ok, I'll delete it.